src/KnowbaseItem.php

```diff
@@ -1458,17 +1458,23 @@ public static function getListRequest(array $params, $type = 'search')
                 if (strlen($params["contains"]) > 0) {
                     $search  = Sanitizer::unsanitize($params["contains"]);
 
-                   // Replace all non word characters with spaces (see: https://stackoverflow.com/a/26537463)
-                    $search_wilcard = preg_replace('/[^\p{L}\p{N}_]+/u', ' ', $search);
-
-                   // Remove last space to avoid illegal syntax with " *"
-                    $search_wilcard = trim($search_wilcard);
-
-                   // Merge spaces since we are using them to split the string later
-                    $search_wilcard = preg_replace('!\s+!', ' ', $search_wilcard);
-
-                    $search_wilcard = explode(' ', $search_wilcard);
-                    $search_wilcard = implode('* ', $search_wilcard) . '*';
+				   // Replace all non word/operator characters with spaces (see: https://stackoverflow.com/a/26537463), ignore those inside quotes
+				    $search_wilcard = preg_replace('/[^\p{L}\p{N}_)(~+\<>*\\\\"-]+(?=([^"]*"[^"]*")*[^"]*$)/u', ' ', $search);
+				   // Remove escape character from quotes
+				    $search_wilcard = preg_replace('![\\\\]["]!u', '"', $search_wilcard);
+				   // Remove all isolated groups of operators, ignore those inside quotes
+				   // i.e. those operators which are neither a prefix nor a suffix for a word, group or phrase
+				    $search_wilcard = preg_replace('!(?<=\s)[~\<>*+-]+(?=\s)(?=([^"]*"[^"]*")*[^"]*$)!u', '', $search_wilcard);
+				   // Remove all * operators immediately following parentheses or quoted terms, ignore those inside quotes
+				    $search_wilcard = preg_replace('!(?<=[)("])[*](?=([^"]*"[^"]*")*[^"]*$)!u', '', $search_wilcard);
+				   // Insert space between closing parentheses and immediately following operator, ignore those inside quotes
+				    $search_wilcard = preg_replace('!(?<=[)])([~\<>+-])(?=([^"]*"[^"]*")*[^"]*$)!u', ' \1', $search_wilcard);
+				   // Replace any * operators immediately following a space, ignore those inside quotes
+				    $search_wilcard = preg_replace('!(?<=\s)[*](?=([^"]*"[^"]*")*[^"]*$)!u', ' ', $search_wilcard);
+				   // Replace groups of ~, >, <, + or - operators with rightmost operator, ignore those inside quotes
+					$search_wilcard = preg_replace('![~\<>+-]+(?=[~\<>+-])(?=([^"]*"[^"]*")*[^"]*$)!u', '', $search_wilcard);
+				   // Merge spaces, ignore those inside quotes
+				    $search_wilcard = preg_replace('!\s+(?=([^"]*"[^"]*")*[^"]*$)!u', ' ', $search_wilcard);
```

Comment:
This piece of code should be moved into a dedicated method, and unit tests should be added on this method. Thus, reading input/result of this method in a test provider may be more clear than reading all these regex patterns.